frontend/javascript/components/moderation/moderation-attachments.vue

```diff
@@ -61,11 +81,25 @@ export default {
       return this.attachments.map((att) => {
         att.url = getUrl(this.config.url.show_attachment, att)
         att.redactUrl = getRedactionUrl(this.config.url.redact_attachment, att)
+        att.messageUrl = getMessageUrl(this.config.url.foimessage, att)
         return att
       })
     }
   },
-  methods: {}
+  methods: {
+    markModerated(att) {
+      postData(
+        getRedactionUrl(this.config.url.mark_attachment_as_moderated, att),
+        {},
+        this.$root.csrfToken
+      ).then(() => {
+        const index = this.attachments.findIndex((x) => x.id === att.id)
```

Comment:
Could this be a `this.attachments = this.attachments.filter(...)`?

Reply:
No, because we cannot directly mutate properties. It seems to be considered a bad practice, can lead to vue missing updated and is forbidden by eslint rule 'vue/no-mutating-props'